debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -352,7 +351,8 @@ protected void generateDeleteRecord(TableId tableId, Object[] oldRowData, Consum
         List<String> columnNames = table.columnNames();
         Object[] values = new Object[messageList.size()];
         messageList.forEach(message -> {
-            int position = columnNames.indexOf(message.getColumnName());
+            final String columnName = message.getColumnName().startsWith("\"") ? message.getColumnName().substring(1, message.getColumnName().length()-1) : message.getColumnName();
```

Comment:
It probably would be worthwhile to extract the de-quoting of the column name into a function so it can be used in both places.